File: Tool/PreUtil/src/output/c/medium/ScoreAfterFlippingMatrix.c
/**https://leetcode.com/problems/score-after-flipping-matrix/ */
//You are given an m x n binary matrix grid.
//A move consists of choosing any row or column and toggling each value in that row or column (i.e., changing all 0's to 1's, and all 1's to 0's).
//Every row of the matrix is interpreted as a binary number, and the score of the matrix is the sum of these numbers.
//Return the highest possible score after making any number of moves (including zero moves).


int matrixScore(int** grid, int gridSize, int* gridColSize){
    int m = gridSize;
    int n = gridColSize[0];
    int* row = (int*)malloc(sizeof(int) * m);
    int* col = (int*)malloc(sizeof(int) * n);
    for(int i = 0; i < m; i++){
        row[i] = 0;
    }
    for(int i = 0; i < n; i++){
        col[i] = 0;
    }
    for(int i = 0; i < m; i++){
        for(int j = 0; j < n; j++){
            if(grid[i][j] == 1){
                row[i]++;
                col[j]++;
            }
        }
    }
    int count = 0;
    for(int i = 0; i < m; i++){
        if(row[i] < n / 2){
            count += n - row[i];
        }
        else{
            count += row[i];
        }
    }
    for(int i = 0; i < n; i++){
        if(col[i] < m / 2){
            count += m - col[i];
        }
        else{
            count += col[i];
        }
    }
    return count;

}

    

File: Tool/PreUtil/src/output/javascript/medium/ScoreAfterFlippingMatrix.js
/**https://leetcode.com/problems/score-after-flipping-matrix/ */
//You are given an m x n binary matrix grid.
//A move consists of choosing any row or column and toggling each value in that row or column (i.e., changing all 0's to 1's, and all 1's to 0's).
//Every row of the matrix is interpreted as a binary number, and the score of the matrix is the sum of these numbers.
//Return the highest possible score after making any number of moves (including zero moves).


/**
 * @param {number[][]} grid
 * @return {number}
 */
var matrixScore = function(grid) {
    let result = 0;
    for(let i = 0; i < grid.length; i++){
        if(grid[i][0] == 0){
            for(let j = 0; j < grid[0].length; j++){
                grid[i][j] = 1 - grid[i][j];
            }
        }
    }
    for(let j = 0; j < grid[0].length; j++){
        let count = 0;
        for(let i = 0; i < grid.length; i++){
            if(grid[i][j] == 1){
                count++;
            }
        }
        if(count < grid.length / 2){
            for(let i = 0; i < grid.length; i++){
                grid[i][j] = 1 - grid[i][j];
            }
        }
    }
    for(let i = 0; i < grid.length; i++){
        for(let j = 0; j < grid[0].length; j++){
            result += grid[i][j] * Math.pow(2, grid[0].length - j - 1);
        }
    }
    return result;
    
};

    